src/hooks/useWebSocket.ts

```diff
@@ -0,0 +1,66 @@
+import type { messageCallbackType, StompHeaders } from "@stomp/stompjs";
+import { Stomp } from "@stomp/stompjs";
+import { useEffect } from "react";
+import SockJS from "sockjs-client";
+import { env } from "~/constants";
+
+/*
+ props로 subscribe method의 Params 를 갖고 오고 싶지만, class method는 type safety하게 접근불가
+ https://stackoverflow.com/questions/70652329/get-return-type-of-class-method-via-method-name-in-typescript
+*/
+
+/*
+@Example
+  useWebSocket({
+    destination: "/asd/asd",
+    callback: () => console.log("소켓값도착")
+  });
+
+  useWebSocket([
+    {
+      destination: "/asd/asd",
+      callback: () => console.log("소켓값도착")
+    },
+    {
+      destination: "/asd/asd2",
+      callback: () => console.log("소켓값도착2")
+    }
+  ]);
+
+
+*/
+const useWebSocket = (
+  subscribes: Util.SingleOrArray<{
+    destination: string;
+    callback: messageCallbackType;
+    headers?: StompHeaders;
+  }>
+) => {
+  useEffect(() => {
+    const sock = new SockJS(`${env.TOPPINGS_SERVER_URL}/stomp/subscribe`);
+    const client = Stomp.over(sock);
+    client.debug = () => {
+      return null;
+    };
+    client.connect({}, () => {
+      if (Array.isArray(subscribes)) {
+        subscribes.forEach(subscribe =>
+          client.subscribe(
+            subscribe.destination,
+            subscribe.callback,
+            subscribe.headers
+          )
```

Comment:
subscribes의 subcribe하고 client의 subscribe하고 네이밍이 중복되서 가독성이 별로라고 느껴집니다!
```suggestion
        subscribes.forEach(({ callback, destination, headers }) =>
          client.subscribe(destination, callback, headers)
        );
```

이 부분은 구조 분해 할당을 통해 네이밍 중복을 피하는 건 어때요?

그리고 궁금한 게 있습니다! 여기 if-else 구문 애들도 사실 한 줄이잖아요? 이럴 경우 중괄호 사용할 지 의논해보고 싶습니다!

저번에 물어봤을 때(회사 플젝)는  한 줄은 괄호 안치신다고 하셨어서 헷갈리네요

Reply:
한줄 선언은 {} 생략 콜!